### PR TITLE
Add solution

### PR DESCRIPTION
**Pull request by IlyaGrynyshyn**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -2,5 +2,14 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_memory = {}
+
+    def wrapper(*args) -> dict:
```
Comment (Dimosphen1) [inline app/main.py]: Change your return type annotation for `wrapper`, it could return `Any` because the function output is not known in advance.

Comment (Dimosphen1) [inline app/main.py]: Consider using a decorator for `wrapper` from `functools` to preserve the wrapped function naming and docstring.

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested.

File: app/main.py [hunk 2]
```diff
@@ -1,6 +1,17 @@
-from typing import Callable
+from typing import Callable, Any
+import functools
```
Comment (Dimosphen1) [inline app/main.py]: `import` should precede `from ... import`, place your import in the right order.

File: app/main.py [hunk 3]
```diff
@@ -1,6 +1,17 @@
...
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    functools.wraps(func)
```
Comment (Dimosphen1) [inline app/main.py]: Place a `wraps` decorator at the right place as [provided in the documentation](https://docs.python.org/uk/3/library/functools.html#functools.wraps).

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested.

File: app/main.py (hunk 2 above)
Comment (Dimosphen1) [reply app/main.py]: Place your imports in the right order according to the previous comment.

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several minor changes were requested.

**Review by viktoria-rybenchuk — APPROVED** [review]
Good job!

**Pull request closed without merging**